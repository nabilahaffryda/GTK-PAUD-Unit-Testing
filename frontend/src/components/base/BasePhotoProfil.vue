<template>
  <div class="box-card">
    <v-btn id="edit-profpic" class="prof-pic grey--text text--darken-2" right icon v-if="isEdit">
      <v-icon large>mdi-camera</v-icon>
    </v-btn>
    <template>
      <v-img :src="image" :aspect-ratio="aspectRatio" />
    </template>
    <profil-picture
      v-if="isEdit"
      :uploadUrl="uploadUrl"
      :useBase64="useBase64"
      trigger="#edit-profpic"
      @upload="upload"
      @uploaded="uploaded"
    />
  </div>
</template>

<script>
import ProfilPicture from '../popup/ProfilPicture';
export default {
  name: 'BasePhotoProfil',
  components: { ProfilPicture },
  props: {
    isEdit: {
      type: Boolean,
      default: false,
    },
    aspectRatio: {
      type: Number,
      default: 4 / 6,
    },
    useBase64: {
      type: Boolean,
      default: false,
    },
    photo: {
      type: String,
      required: true,
    },
    photodef: {
      type: String,
      default: 'avatar.png',
    },
    uploadUrl: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      image: '',
    };
  },
  mounted() {
    this.checkImage(this.photo);
  },
  methods: {
    upload(data) {
      this.$emit('upload', data);
    },
    uploaded() {
      this.$success('Unggah Foto berhasil disimpan');
      this.$emit('reload');
    },
    checkImage(imageSrc) {
      let img = new Image();
      img.onload = () => {
        this.image = imageSrc;
      };
      img.onerror = () => {
        this.image = this.$imgUrl(this.photodef);
      };
      img.src = imageSrc;
    },
  },
  watch: {
    photo: 'checkImage',
  },
};
</script>
<style scoped>
.box-card {
  position: relative;
}
.prof-pic {
  position: absolute;
  z-index: 1;
  right: 3%;
  top: 3%;
}
</style>
