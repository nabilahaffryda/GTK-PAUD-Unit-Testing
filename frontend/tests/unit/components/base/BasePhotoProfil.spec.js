import { mount, createLocalVue } from "@vue/test-utils";
import BasePhotoProfil from "@/components/base/BasePhotoProfil.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BasePhotoProfil.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    test('render photo profil', () => {
        const VPP = {
            props: [
                'isEdit',
                'aspectRatio',
                'useBase64',
                'photo',
                'photodef',
                'uploadUrl',
                'useTrigger',
                'useBtn'
            ],
            template: '<div><slot :uploadUrl="uploadUrl" /></div>'
        }

        const wrapper = mount(BasePhotoProfil, {
            localVue,
            vuetify,
            stubs: {
                VPP
            },
            propsData: {
                photo: 'avatar.png'
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})