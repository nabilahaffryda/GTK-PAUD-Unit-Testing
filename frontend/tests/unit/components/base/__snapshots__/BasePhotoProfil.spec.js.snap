// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePhotoProfil.vue render photo profil 1`] = `
<div class="box-card">
  <!---->
  <div class="v-image v-responsive theme--light">
    <div class="v-responsive__sizer" style="padding-bottom: 150%;"></div>
    <div class="v-responsive__content"></div>
  </div>
  <!---->
</div>
`;
